### ISSUE TITLE
Verification if remote branch is up to date fails

### ISSUE THREAD

**Issue by pvdlg** [semantic-release/semantic-release#1062]
Ref https://github.com/semantic-release/semantic-release/issues/563#issuecomment-451754756

I had another of the `! [rejected] HEAD -> beta (non-fast-forward)` errors. We are using the latest beta of `semantic-release`. I’m now fairly sure that it’s a timing problem. I merged several pull requests in quick succession, I assume that while the release build started, I merged another pr, hence the `non-fast-forward` error. 

I wonder if we could add an error handler for that error to check if the last remote commit changed and if it did, log a more helpful message and then exit without an error, as we can assume that the new commit triggered another build?

https://travis-ci.org/nock/nock/jobs/475937483#L505

```
[10:13:48 AM] [semantic-release] › ✔  Run automated release from branch beta
2019-01-06T10:13:48.430Z semantic-release:git Error: Command failed: git push --dry-run https://github.com/nock/nock.git HEAD:beta
remote: Invalid username or password.
fatal: Authentication failed for 'https://github.com/nock/nock.git/'
    at makeError (/home/travis/build/nock/nock/node_modules/semantic-release/node_modules/execa/index.js:174:9)
    at Promise.all.then.arr (/home/travis/build/nock/nock/node_modules/semantic-release/node_modules/execa/index.js:278:16)
    at process._tickCallback (internal/process/next_tick.js:68:7)
2019-01-06T10:13:48.836Z semantic-release:git Error: Command failed: git push --dry-run https://[secure]@github.com/nock/nock.git HEAD:beta
To https://github.com/nock/nock.git
 ! [rejected]        HEAD -> beta (non-fast-forward)
```

_Originally posted by @gr2m in https://github.com/semantic-release/semantic-release/issues/563#issuecomment-451754756_

**Comment by Berkmann18** [comment]
So why did it worked from time time with the same config?

**Comment by pvdlg** [comment]
@Berkmann18 what do you mean?

**Comment by Berkmann18** [comment]
@pvdlg Basically, having the same configuration, @semantic-release-bot sometimes *did* write the changes but not always which is a bit strange.

**Comment by pvdlg** [comment]
Hopefully fixed by #1383. Unfortunately I can't reproduce the exact case from https://github.com/semantic-release/semantic-release/issues/563#issuecomment-451754756
